src/taskEvents/script.js

```diff
@@ -0,0 +1,283 @@
+import {
+  getTaskLogs,
+  getUserData,
+  getSelfDetails,
+  createElement,
+  getData,
+  addLoader,
+  addErrorMessage,
+  removeLoader,
+} from './utils.js';
+
+import { createEventCard } from './logCard.js';
+
+const container = document.getElementById('task-events-container');
+const modal = document.getElementById('modal');
+const overlay = document.querySelector('.overlay');
+
+const closeBtn = document.getElementById('close-btn');
+closeBtn.addEventListener('click', closeModal);
+const closeBtn2 = document.getElementById('close-btn2');
+closeBtn2.addEventListener('click', closeGenericModal);
+
+async function createProfileModal(username) {
+  try {
+    overlay.style.display = 'block';
+    const { user } = await getUserData(username);
+    // another call for roles will be made when we have userSkills collection
+    const userImg = createElement({
+      type: 'img',
+      attributes: { src: user.picture.url, alt: 'user img', class: 'user-img' },
```

Comment:
**user.picture.url** 
What will happen here if a user is undefined (service error) ?